src/api.ts

```diff
@@ -325,6 +325,63 @@ type CreateAttachmentParams = {
   metadata?: Maybe<Record<string, any>>;
 };
 
+export class SharedCache {
+  private static instance: SharedCache | null = null;
+  private cache: Map<string, any>;
+
+  private constructor() {
+    this.cache = new Map();
+  }
+
+  static getInstance(): SharedCache {
+    if (!SharedCache.instance) {
+      SharedCache.instance = new SharedCache();
+    }
+    return SharedCache.instance;
+  }
+
+  public getPromptCacheKey(
```

Comment:
Now that the cache is not prompt only, let's not leak that business logic in the cache layer.